src/napari_spatialdata/_view.py

```diff
@@ -119,10 +119,15 @@ def _on_selection(self, event: Optional[Any] = None) -> None:
     def _select_layer(self) -> None:
         """Napari layers."""
         layer = self._viewer.layers.selection._current
+        self.model.layer = layer
         if not isinstance(layer.metadata.get("adata", None), AnnData):
-            raise NotImplementedError(":class:`anndata.AnnData` not found in any `layer.metadata`.")
+            if hasattr(self, "x_widget"):
+                self.x_widget.clear()
+                self.y_widget.clear()
+                self.color_widget.clear()
+            logger.warning(":class:`anndata.AnnData` not found in any `layer.metadata`.")
```

Comment:
can you remove the warning? cause it's not necessarily wrong loading an image with the view widget